File: Data Structures & Algorithms/two-integer-sum/submission-5.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> indicesMap = new HashMap<Integer, Integer>();

        for(int i = 0; i < nums.length; i ++){
            if(indicesMap.containsKey(target - nums[i])){
                return new int[]{indicesMap.get(target - nums[i]) , i};
            } else {
                indicesMap.put(nums[i], i);
            }
        }
        return new int[] {};
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        seen = {}

        for i in range(len(numbers)):
            y = target - numbers[i]
            if y in seen.keys():
                return [seen[y] + 1, i + 1]
            else:
                seen[numbers[i]] = i

        return []	



        